sys/rtc_utils/rtc_utils.c

```diff
@@ -258,3 +259,23 @@ bool rtc_tm_valid(const struct tm *t)
     rtc_tm_normalize(&norm);
     return rtc_tm_compare(t, &norm) == 0;
 }
+
+#ifdef MODULE_PERIPH_RTC_SETTER_CALLBACK
+int rtc_set_time_with_callback(struct tm *time)
```

Comment:
Hmm, I wonder if it wouldn't be better to rename `rtc_set_time()` to e.g. `rtc_set_time_dev()` or whatever. That could also be extended to directly provide the previous values (including the ms parameter) for improved robustness. And `rtc_set_time()` would than be a common wrapper on top, which either just replicates the old behavior (if module `periph_rtc_setter_callback` is not used), or calls the callback.

I fear that this feature might be a footgun if one has to ensure that all calls of `rtc_set_time()` will be replaced with `rtc_set_time_with_callback()` to work correctly.

I am aware that this would involve touching all RTC implementations...

Reply:
I completely agree with you on the footgun. I just wanted to reduce my impact on code with this second proposal. I am willing to rework this PR accordingly but it might take some weeks until I get back to it as I am currently under a relatively high load.